src/App.js

```diff
@@ -1,25 +1,82 @@
+/* eslint-disable no-console */
 import React from 'react';
 
 import './App.scss';
+import { Clock } from './Clock';
 
-const App = () => {
-  setInterval(() => {
-    const date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+class App extends React.Component {
+  state = {
+    isClockVisible: true,
+    clockName: 1,
+  }
+
+  render() {
+    const { isClockVisible } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        <p>
+          Current time:
+          {' '}
+          <span className="clock">
+            <Clock isClockVisible={isClockVisible} />
+          </span>
+        </p>
+
+        <button
+          type="button"
+          onClick={() => {
+            if (!isClockVisible) {
+              const clock = document.querySelector('.clock');
+
+              clock.hidden = false;
+              this.setState({
+                isClockVisible: true,
+              });
+            }
+          }}
+        >
+          Show Clock
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            if (isClockVisible) {
+              const clock = document.querySelector('.clock');
+
+              clock.hidden = true;
+              this.setState({
+                isClockVisible: false,
+              });
+            }
+          }}
+        >
+          Hide Clock
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            this.setState((prevState) => {
+              const newClockName = Math.round(Math.random() * 100);
```

Comment:
![image](https://user-images.githubusercontent.com/38065883/122362772-15d52680-cf61-11eb-803d-a14d57cd653f.png)
You ain't passing name as prop to the Clock component

Reply:
Use componentDidUpdate in Clock component to log new clock's name